src/main/java/com/code_intelligence/jazzer/mutation/combinator/MutatorCombinators.java

```diff
@@ -179,6 +189,57 @@ public R detach(R value) {
     };
   }
 
+  public static <T, @ImmutableTypeParameter R> SerializingMutator<R> mutateThenMapToImmutable(
+      SerializingMutator<T> mutator, Function<T, R> map, Function<R, T> inverse,
+      Function<Predicate<Debuggable>, String> debug) {
+    return new PostComposedMutator<T, R>(mutator, map, inverse) {
+      @Override
+      public R detach(R value) {
+        return value;
+      }
+
+      @Override
+      public String toDebugString(Predicate<Debuggable> isInCycle) {
+        return debug.apply(isInCycle);
+      }
+    };
+  }
+
+  public static SerializingMutator<Integer> mutateIndices(int length) {
```

Comment:
`mutateIndices` basically returns a bounded `IntegralMutator`, doesn't it? I wonder if we could reuse that, or the integral mutation part, here as well.

Reply:
I was thinking the same when I saw the PR that introduced IntegralMutator! However, after looking at `bitFlip` and `randomWalk` I am not sure how useful that is for enums. Should keep this in mind until we have a benchmark ready to test the effectivenes of each mutator?

Reply:
Sounds good.